scripts/setup-fedora.sh

```diff
@@ -0,0 +1,244 @@
+#!/bin/bash
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script documents setting up a Fedora 40 host for Velox
+# development.  Running it should make you ready to compile.
+#
+# Environment variables:
+# * INSTALL_PREREQUISITES="N": Skip installation of packages for build.
+# * PROMPT_ALWAYS_RESPOND="n": Automatically respond to interactive prompts.
+#     Use "n" to never wipe directories.
+#
+# You can also run individual functions below by specifying them as arguments:
+# $ scripts/setup-fedora.sh install_googletest install_fmt
+#
+
+set -efx -o pipefail
+# Some of the packages must be build with the same compiler flags
+# so that some low level types are the same size. Also, disable warnings.
+SCRIPTDIR=$(dirname "${BASH_SOURCE[0]}")
+source $SCRIPTDIR/setup-helper-functions.sh
+SUDO="${SUDO:-"sudo --preserve-env"}"
+
+NPROC=$(getconf _NPROCESSORS_ONLN)
+export CXXFLAGS=$(get_cxx_flags) # Used by boost. 
+export CFLAGS=${CXXFLAGS//"-std=c++17"/} # Used by LZO.
+CMAKE_BUILD_TYPE="${BUILD_TYPE:-Release}"
+DEPENDENCY_DIR=${DEPENDENCY_DIR:-$(pwd)}
+BUILD_DUCKDB="${BUILD_DUCKDB:-true}"
+USE_CLANG="${USE_CLANG:-false}"
+
+FB_OS_VERSION="v2024.05.20.00"
```

Comment:
This was updated not that long ago. The current version is `v2024.07.01.00`. Please see the other setup files.

Reply:
Yes, it looks like there were a number of changes to the setup scripts that I missed in my rebase.  I just updated to account for the changes.

I've only tested with Clang 15, ie.
`USE_CLANG=true ./setup-fedora.sh`

After running the setup script, building with clang required once additional import fix (exec/QueryTraceConfig.h). I'm not sure why that issue isn't showing up elsewhere.  But with that in place, I am able to build Velox (`make debug && make unittest`).

I have not tested with GCC 13 to see if that works with this.  If there are known build issue with it, then I would guess not.

Reply:
> One of the issues was that it wasn't the dependencies but Velox that would not compile with the system GCC version (it being pretty new).

That was probably #5898